crates/sqlexec/src/parser.rs

```diff
@@ -127,17 +128,37 @@ impl fmt::Display for DropDatabaseStmt {
 }
 
 #[derive(Debug, Clone, PartialEq, Eq)]
-pub struct AlterDatabaseRenameStmt {
+pub enum AlterDatabaseOperation {
+    RenameDatabase { new_name: Ident },
+    GrantAccess { permission: OperationPermission },
+    RevokeAccess { permission: OperationPermission },
+}
+
+impl fmt::Display for AlterDatabaseOperation {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::RenameDatabase { new_name } => {
+                write!(f, "RENAME TO {new_name}")
+            }
+            Self::GrantAccess { permission } => {
+                write!(f, "GRANT ACCESS TO {permission}")
+            }
+            Self::RevokeAccess { permission } => {
+                write!(f, "REVOKE ACCESS TO {permission}")
+            }
+        }
+    }
+}
```

Comment:
Was there an existing system that you based the `REVOKE`/`GRANT` format on? Typically when I see these, I associated permissions for a user and not the database object itself. The way I see this, we're setting a property on an object that applies to everybody accessing that object.

When setting the property of an object, I'd expect to see something more along the lines of:

```
ALTER DATABASE <database> SET WRITE_MODE READONLY;
ALTER DATABASE <database> SET WRITE_MODE WRITE_DML;
```

(or similar)

(also open to changing my mind)

Reply:
I like `SET` / `UNSET` as `GRANT` / `REVOKE`.

What is the default? Is _that_ settable somewhere on the deployment?

Reply:
Yeah, I guess `SET`/`UNSET` makes more sense in this context.

> I like SET / UNSET as GRANT / REVOKE.

@tychoish A bit unclear, do you prefer `SET`/`UNSET` or `GRANT`/`REVOKE`?

> What is the default? Is that settable somewhere on the deployment?

The default for native and temporary tables is both read and write permissions. The default for external sources is read only.

Reply:
Prefer `SET` / `UNSET`

We should make sure to document these defaults.